packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -235,6 +235,28 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
         };
     }
 
+    /**
+     * {@inheritDoc (ISharedObject:interface).getGCData}
+     */
+    public getGCData(): IGCData {
+        this._isSummarizing = true;
+         // Create a SummarySerializer that will be used to serialize IFluidHandles in this object. SummarySerializer
+         // tracks the routes of handles that it serializes. These represent outbound routes to other Fluid object.
+        const serializer = new SummarySerializer(this.runtime.channelsRoutingContext);
+
+        // Take a snapshot of the object's data which will serialize all the handles.
+        // Note: Using the snapshotCore API during GC is a little expensive but its okay for now.
+        this.snapshotCore(serializer);
+
+        this._isSummarizing = false;
```

Comment:
I'd add assert above it that it was still true before we reset. Similar (but reverse) at beginning of function - these asserts will catch reentrancy if it were to happen.